### PR TITLE
Mbarba/dump events

### PR DESCRIPTION
**Pull request by MatBarba**
Integrate the stable ids event dump in the dumper pipeline. The ids history will be dumped automatically if there is any, in the BRC4 expected format.
Also if there is no files created, then don't create a manifest or folder (e.g. when dumping history only, and some dbs don't have any).

### CODE REVIEW

File: lib/perl/Bio/EnsEMBL/Pipeline/Runnable/BRC4/Manifest.pm [hunk 1]
```diff
@@ -75,6 +79,26 @@ sub run {
...
 }
 
+sub manifest_has_files {
+  my ($manifest) = @_;
+
+  my $has_files = 0;
+  for my $name (keys %$manifest) {
+
+    if (ref($manifest->{$name}) eq 'HASH') {
+      foreach my $subname (sort keys %{ $manifest->{$name} }) {
```
Comment (JAlvarezJarreta) [inline lib/perl/Bio/EnsEMBL/Pipeline/Runnable/BRC4/Manifest.pm:89]: Why `sort`? I do not see the value on sorting the keys.

File: lib/python/ensembl/brc4/runnable/dump_stable_ids.py [hunk 2]
```diff
@@ -215,6 +216,19 @@ def _name_event(self) -> None:
...
+        """Remove the empty old pairs when the event is not 'new'."""
+        if not self.name:
+            self._name_event()
+        
+        if self.name != "new":
+            new_pairs = []
+            for pair in self.pairs:
+                if pair["old_id"] == "":
+                    continue
+                new_pairs.append(pair)
```
Comment (JAlvarezJarreta) [inline lib/python/ensembl/brc4/runnable/dump_stable_ids.py:229]: This is the only way I can think that justifies the need of `continue`:
```suggestion
                if ("old_id" in pair) and (pair["old_id"] == ""):
                    continue
                new_pairs.append(pair)
```

File: schema/manifest_schema.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
         "seq_attrib" : { "$ref" : "#/definitions/file_info" },
         "genome" : { "$ref" : "#/definitions/file_info" },
         "functional_annotation" : { "$ref" : "#/definitions/file_info" },
-        "agp" : { "$ref" : "#/definitions/agp_info" }
+        "agp" : { "$ref" : "#/definitions/agp_info" },
+        "events" : { "$ref" : "#/definitions/file_info" }
```
Comment (JAlvarezJarreta) [inline schema/manifest_schema.json:19]: This is the most annoying thing for me from JSON: not being able to finish the last element with `,` so the new additions only cover new lines.

**Review by JAlvarezJarreta — APPROVED** [review]
I find it surprising that there is no runnable like `ExtractConnection.pm` in Hive :exploding_head: 

_Note:_ -1 point for adding a new Perl file to the codebase though :stuck_out_tongue_winking_eye:

File: lib/python/ensembl/brc4/runnable/dump_stable_ids.py (hunk 2 above)
Comment (MatBarba) [reply lib/python/ensembl/brc4/runnable/dump_stable_ids.py]: Replaced with .get()

File: schema/manifest_schema.json (hunk 3 above)
Comment (MatBarba) [reply schema/manifest_schema.json:19]: Good idea to add a json checker to our CI/CD then?

Comment (JAlvarezJarreta) [reply schema/manifest_schema.json:19]: Yes, this is already planned once I finally sit down and push on the CI/CD documentation.

**Pull request merged**